megamek/src/megamek/client/ui/swing/unitSelector/TWAdvancedSearchPanel.java

```diff
@@ -0,0 +1,1242 @@
+        cbxEnableArmorSearch.addItemListener(this);
+
+        for (int i = 1; i <= 20; i++) {
+            cboQty.addItem(Integer.toString(i));
+        }
+        cboQty.setSelectedIndex(0);
+
+        //Setup table filter combo boxes
+        DefaultComboBoxModel<String> unitTypeModel = new DefaultComboBoxModel<>();
+        unitTypeModel.addElement(Messages.getString("MechSelectorDialog.All"));
+        unitTypeModel.addElement(UnitType.getTypeDisplayableName(UnitType.MEK));
+        unitTypeModel.addElement(UnitType.getTypeDisplayableName(UnitType.TANK));
+        unitTypeModel.addElement(UnitType.getTypeDisplayableName(UnitType.BATTLE_ARMOR));
+        unitTypeModel.addElement(UnitType.getTypeDisplayableName(UnitType.INFANTRY));
+        unitTypeModel.addElement(UnitType.getTypeDisplayableName(UnitType.PROTOMEK));
+        unitTypeModel.addElement(UnitType.getTypeDisplayableName(UnitType.AERO));
+        unitTypeModel.setSelectedItem(Messages.getString("MechSelectorDialog.All"));
+
+        cboUnitType.setModel(unitTypeModel);
+        cboUnitType.addActionListener(this);
+
+        DefaultComboBoxModel<String> techLevelModel = new DefaultComboBoxModel<>();
+        for (int i = 0; i < TechConstants.SIZE; i++) {
+            techLevelModel.addElement(TechConstants.getLevelDisplayableName(i));
+        }
+        techLevelModel.setSelectedItem(TechConstants.getLevelDisplayableName(TechConstants.SIZE-1));
+        cboTechLevel.setModel(techLevelModel);
+        cboTechLevel.addActionListener(this);
+
+        DefaultComboBoxModel<String> techClassModel = new DefaultComboBoxModel<>();
+        techClassModel.addElement("All");
+        techClassModel.addElement("Inner Sphere");
+        techClassModel.addElement("Clan");
+        techClassModel.addElement("IS/Clan");
+        techClassModel.addElement("(Unknown Technology Base)");
+        techClassModel.setSelectedItem("All");
+        cboTechClass.setModel(techClassModel);
+        cboTechClass.addActionListener(this);
+
+
+        //Setup Weapons Table
+        scrTableWeapons.setMinimumSize(new Dimension(850, 150));
+        scrTableWeapons.setPreferredSize(new Dimension(850, 150));
+        weaponsModel = new WeaponsTableModel();
+        tblWeapons = new MegamekTable(weaponsModel, WeaponsTableModel.COL_NAME);
+        TableColumn wpsCol = tblWeapons.getColumnModel().getColumn(
+                WeaponsTableModel.COL_QTY);
+        wpsCol.setCellEditor(new DefaultCellEditor(cboQty));
+        tblWeapons.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        weaponsSorter = new TableRowSorter<>(weaponsModel);
+        tblWeapons.setRowSorter(weaponsSorter);
+        tblWeapons.addKeyListener(this);
+        TableColumn column = null;
+        for (int i = 0; i < WeaponsTableModel.N_COL; i++) {
+            column = tblWeapons.getColumnModel().getColumn(i);
+            if ((i == WeaponsTableModel.COL_QTY)) {
```

Comment:
This would be better handled as a switch.

Reply:
Would make it two lines per case with the break. Not really an improvement imo